cdap-client/src/main/java/co/cask/cdap/client/ProgramClient.java

```diff
@@ -485,24 +452,25 @@ public void setServiceInstances(String appId, String serviceId, int instances)
    * @param appId ID of the application that the program belongs to
    * @param programType type of the program
    * @param programId ID of the program
+   * @param state - filter by status of the program
    * @return the run records of the program
    * @throws IOException if a network error occurred
    * @throws NotFoundException if the application or program could not be found
    * @throws UnauthorizedException if the request is not authorized successfully in the gateway server
    */
-  public List<RunRecord> getAllProgramRuns(String appId, ProgramType programType, String programId,
-                                           long startTime, long endTime, int limit)
+  public List<RunRecord> getProgramRuns(String appId, ProgramType programType, String programId, String state,
```

Comment:
Should perhaps use ``ProgramRunStatus`` instead of ``String`` for ``state``, but since you brought up that that would be an API change, maybe later.

Similar comment about using ``Id.Program`` instead of the first 3 parameters.